src/controller/java/src/chip/devicecontroller/ChipDeviceController.java

```diff
@@ -488,11 +488,34 @@ public boolean openPairingWindowWithPINCallback(
         deviceControllerPtr, devicePtr, duration, iteration, discriminator, setupPinCode, callback);
   }
 
-  /* Shutdown all cluster attribute subscriptions for a given device */
-  public void shutdownSubscriptions(long devicePtr) {
-    shutdownSubscriptions(deviceControllerPtr, devicePtr);
+  public int getFabricIndex() {
+    return getFabricIndex(deviceControllerPtr);
   }
 
+  /* Shutdown all subscriptions */
+  /* if schedule is true, it would be forced to run in matter thread */
+  public void shutdownSubscriptions(boolean schedule) {
+    shutdownSubscriptions(deviceControllerPtr, null, null, null, schedule);
+  }
+
+  /* Shutdown the subscriptions based upon fabricIndex,  */
+  /* if schedule is true, it would be forced to run in matter thread */
+  public void shutdownSubscriptions(Integer fabricIndex, boolean schedule) {
+    shutdownSubscriptions(deviceControllerPtr, fabricIndex, null, null, schedule);
+  }
+
+  /* Shutdown the subscriptions based upon peerNodeId, fabricIndex */
+  /* if schedule is true, it would be forced to run in matter thread */
+  public void shutdownSubscriptions(Integer fabricIndex, Long peerNodeId, boolean schedule) {
+    shutdownSubscriptions(deviceControllerPtr, fabricIndex, peerNodeId, null, schedule);
+  }
+
+  /* Shutdown the subscriptions based upon peerNodeId, fabricIndex, and subscriptionId */
+  /* if schedule is true, it would be forced to run in matter thread */
+  public void shutdownSubscriptions(
```

Comment:
```
/**
 * Closes a specific subscription for a node.
 *
 * @param fabricIndex the fabric index of to which the node belongs
 * @param peerNodeId the node ID of the device for which the subscription belongs
 * @param subscriptionId the ID of the subscription on the node which should be canceled
 * @param schedule if true, asynchronously schedules a shutdown instead of blocking
 */
```